cdap-examples/WikipediaPipeline/src/main/scala/co/cask/cdap/examples/wikipedia/ScalaSparkLDA.scala

```diff
@@ -38,10 +38,12 @@ class ScalaSparkLDA extends SparkMain {
     val sc = new SparkContext
 
     val arguments = sec.getRuntimeArguments
+    val namespace = arguments.get(WikipediaPipelineApp.NAMESPACE_ARG)
 
     // Pre-process data for LDA
     val (corpus, vocabArray, _) = ClusteringUtils.preProcess(
-      sc.fromDataset(WikipediaPipelineApp.NORMALIZED_WIKIPEDIA_DATASET), sec.getRuntimeArguments.toMap)
+      if (namespace != null) sc.fromDataset(namespace, WikipediaPipelineApp.NORMALIZED_WIKIPEDIA_DATASET)
```

Comment:
don't do if else thing. Just like previous comment if namespace args in runtime is null assign namespace with current namespace of the program